File: frontend/src/types/menu.ts
export interface MenuDish {
  id: string
  dishName: string
  description: string | null
  displayOrder: number
}

export interface Menu {
  id: string
  restaurantId: string
  name: string
  description: string | null
  displayOrder: number
  dishes: MenuDish[]
  createdAt: string
}

export interface MenuDishRequest {
  dishName: string
  description?: string
  displayOrder: number
}

export interface MenuRequest {
  name: string
  description?: string
  dishes: MenuDishRequest[]
}

File: backend/src/main/java/com/bwevent/domain/model/Menu.java
package com.bwevent.domain.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.time.Instant;
import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

/** A restaurant-wide preset menu (e.g. "Menu Découverte") — not owned by any
 * particular event type, since the same menu is often reused across several. */
@Entity
@Table(name = "menus")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Menu {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @Column(name = "restaurant_id", nullable = false)
    private UUID restaurantId;

    @Column(nullable = false)
    private String name;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(name = "display_order", nullable = false)
    @Builder.Default
    private Integer displayOrder = 0;

    @OneToMany(mappedBy = "menu", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.EAGER)
    @OrderBy("displayOrder ASC")
    @Builder.Default
    private List<MenuDish> dishes = new ArrayList<>();

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private Instant createdAt;
}
